### PR TITLE
[Refactor Plasma Prototype] This is a prototype to show case end goal of Plasma refactoring

### PR DESCRIPTION
**Pull request by scv119**
This is a serials of PR that refactors PlasmaStore with following goals:
1. Separate of the concerns of object eviction/garbage collection from object storage.
2. OOPfy classes to make them testable.

This PR is just moving things around and renaming things. Specifically:
We separate ObjectTable from PlasmaStoreInfo, and rename PlasmaStoreInfo to PlasmaStoreConfig. The idea is PlasmaStoreConfig stores static configurations for Plasma, 
where as the ObjectTable as the most important member in plasma, stores the allocations info for all objects stores in Plasma

[Next PR](https://github.com/scv119/ray/pull/13)

### CODE REVIEW

File: src/ray/object_manager/plasma/quota_aware_policy.cc [hunk 1]
```diff
@@ -26,8 +26,8 @@
 
 namespace plasma {
 
-QuotaAwarePolicy::QuotaAwarePolicy(PlasmaStoreInfo *store_info, int64_t max_size)
-    : EvictionPolicy(store_info, max_size) {}
+QuotaAwarePolicy::QuotaAwarePolicy(const ObjectTable &object_table, int64_t max_size)
```
Comment (ericl) [inline src/ray/object_manager/plasma/quota_aware_policy.cc:29]: Side note: we should remove QuotaAwarePolicy (it's deprecated / unused).

**Review by ericl — APPROVED** [review]
Seems windows build has an issue: src/ray/object_manager/plasma/store.cc(146): error C7555: use of designated initializers requires at least '/std:c++latest'

File: src/ray/object_manager/plasma/plasma.h [hunk 2]
```diff
@@ -65,33 +65,31 @@ enum class ObjectStatus : int {
...
-  /// Objects that are in the Plasma store.
-  ObjectTable objects;
+/// The plasma store config.
+struct PlasmaStoreConfig {
   /// Boolean flag indicating whether to start the object store with hugepages
   /// support enabled. Huge pages are substantially larger than normal memory
   /// pages (e.g. 2MB or 1GB instead of 4KB) and using them can reduce
   /// bookkeeping overhead from the OS.
-  bool hugepages_enabled;
+  const bool hugepages_enabled;
```
Comment (rkooo567) [inline src/ray/object_manager/plasma/plasma.h:74]: I think this is not used anymore. We can probably delete it? cc @ericl

**Review by rkooo567 — APPROVED** [review]
Nice change! Btw, is there a document that organizes the plans now?

**Comment by suquark** [comment]
I'll review it when there are further progress.

**Comment by scv119** [comment]
yeah please ignore the activities in this PR for now.

**Comment by scv119** [comment]
close in favor of this stack: #17344

**Pull request closed without merging**